### ISSUE TITLE
opcodes are wrong in dns.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2945]
|<img alt="yg10's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| yg10 reported|
|-|-|
|Trac ID|trac#2945|
|Type|enhancement|
|Created|2007-12-11 02:03:38Z|
|Branch|https://github.com/twisted/twisted/tree/dns-notify-update-2945|

in dns.py OP_NOTIFY has value 3. This value is reserved, correct value for OP_NOTIFY is 4, also according to RFC 2136 OP_UPDATE (5) was added.

Tiny patch below should fix that

--- dns.py.orig Mon Dec 10 17:40:40 2007
+++ dns.py      Mon Dec 10 17:42:59 2007
@@ -132,7 +132,7 @@

 # Opcodes
-OP_QUERY, OP_INVERSE, OP_STATUS, OP_NOTIFY = range(4)
+OP_QUERY, OP_INVERSE, OP_STATUS, OP_RESERVED, OP_NOTIFY, OP_UPDATE = range(6)

 # Response Codes
 OK, EFORMAT, ESERVER, ENAME, ENOTIMP, EREFUSED = range(6)

Attachments:

* [dns.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/2a8/2a8a15f1fccbf07279ef24c839182d5f102cdb20/fff979f0156e984c6a6140e178fbd9fe4830e676.diff) (325 bytes) - added by yg10 on 2007-12-11 02:04:16Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2945 2945
type__enhancement enhancement
reporter__yg10 yg10
priority__highest highest
milestone__ 
branch__branches_dns_notify_update_2945 branches/dns-notify-update-2945
branch_author__ 
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1197338618000000 1197338618000000
changetime__1197376255000000 1197376255000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [22064]) Branching to 'dns-notify-update-2945'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [22065]) unit test for some cases of DNSServerFactory, fix the value of OP_NOTIFY and add a definition for OP_UPDATE

refs [#2945](https://github.com/twisted/twisted/issues/2945)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

I also fixed a duplicate test method name in test_names, so an existing test which was never run before is now running.

Ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

One thing a bit disturbing is that test_updateMessageReceived passes even without the patch in dns.py. Maybe you could add a simple check OP_UPDATE == 5?

Otherwise, looks good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @therve|
|-|-|

but I don't want to implement handleUpdate now.  I could delete the definition of OP_UPDATE?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @exarkun|
|-|-|

Fine enough, I guess you can leave OP_UPDATE like this, just open a ticket for the implementation of handleUpdate. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [22066]) Merge dns-notify-update-2945

Author: exarkun
Reviewer: therve
Fixes [#2945](https://github.com/twisted/twisted/issues/2945)
Refs [#2946](https://github.com/twisted/twisted/issues/2946)

Fix the value of OP_NOTIFY in twisted.names.dns and add a definition for
OP_UPDATE.  Add unit tests for the top-level handling of messages with
these opcodes and fix an unrelated incorrectly named/defined test method
so that it doesn't occult another test method.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
